client/components/Editor/index.jsx

```diff
@@ -48,10 +61,39 @@ class Editor extends Component {
 
     onClickScenarioAction(event, data) {
         if (data.name === 'save-scenario') {
-            // TODO: Determine if we need to actually
-            //       save the scenario, or if the
-            //       auto-save is sufficient.
-            this.updateEditorMessage('Teacher Moment saved');
+            this.updateScenario();
+        }
+
+        if (data.name === 'save-status') {
+            const scenario = Object.assign(
+                {},
+                {
+                    ...this.state.scenario,
+                    status: data.id
+                }
+            );
+
+            this.setState({ scenario }, () => {
```

Comment:
Where we're using `setState` for the scenario, should we use Redux and the `setScenario` action instead? At the moment it seems like we have two state systems going for storing scenario information. Let me know if I'm missing why we'd want to keep them separate though.